### PR TITLE
docs: add mathematical formulas to metric docstrings

### LINKED ISSUE #11: Add eval metric formulas to docs
I believe Sphinx can render latex using this directive:

```
.. math::
    -\frac{1}{N} \sum_{i=1}^{N} \left[ y_i \log(p_i) + (1 - y_i) \log(1 - p_i) \right]
```

It would be nice to add the mathematical definition of each metric to its API doc.

### PR DESCRIPTION
**Pull request by nikolasavic3**
This PR adds mathematical definitions to the metric class docstrings using Sphinx LaTeX directives.

I am attaching a pdf of the website to show how the documentation renders the math formulas.
[metrax — metrax documentation.pdf](https://github.com/user-attachments/files/18814751/metrax.metrax.documentation.pdf)

Fixes #11

### CODE REVIEW

File: src/metrax/metrics.py [hunk 1]
```diff
@@ -49,7 +49,24 @@ def _divide_no_nan(x: jax.Array, y: jax.Array) -> jax.Array:
 
 @flax.struct.dataclass
 class MSE(clu_metrics.Average):
-  """Computes the mean squared error for regression problems given `predictions` and `labels`."""
+  r"""Computes the mean squared error for regression problems given `predictions` and `labels`.
+
```
Comment (jeffcarp) [inline src/metrax/metrics.py:53]: Nit: could you remove any leading whitespace in the docstrings (on this line and any below)?

i.e.:
image

File: src/metrax/metrics.py [hunk 2]
```diff
@@ -284,7 +366,29 @@ def compute(self) -> jax.Array:
 
 @flax.struct.dataclass
 class AUCPR(clu_metrics.Metric):
-  """Computes area under the precision-recall curve for binary classification given `predictions` and `labels`.
+  r"""Computes area under the precision-recall curve for binary classification given `predictions` and `labels`.
+
+  The Precision-Recall curve shows the tradeoff between precision and recall at different
+  classification thresholds. The area under this curve (AUC-PR) provides a single score
+  that represents the model's ability to distinguish between classes across all possible
+  thresholds.
```
Comment (jeffcarp) [inline src/metrax/metrics.py:374]: Can this have slightly more detail to distinguish it from the docstring in AUCROC? Maybe like:

`... provides a single score that represents the model's ability to identify positive cases across all possible classification thresholds, particularly in imbalanced datasets.`

File: src/metrax/metrics.py [hunk 3]
```diff
@@ -448,7 +552,26 @@ def compute(self) -> jax.Array:
 
 @flax.struct.dataclass
 class AUCROC(clu_metrics.Metric):
-  """Computes area under the receiver operation characteristic curve for binary classification given `predictions` and `labels`.
+  r"""Computes area under the receiver operation characteristic curve for binary classification given `predictions` and `labels`.
+
+  The ROC curve shows the tradeoff between the true positive rate (TPR) and false positive
+  rate (FPR) at different classification thresholds. The area under this curve (AUC-ROC)
+  provides a single score that represents the model's ability to distinguish between classes
+  across all possible thresholds.
```
Comment (jeffcarp) [inline src/metrax/metrics.py:560]: For this one, maybe an extra detail:

`... represents the model's ability to discriminate between positive and negative cases across all possible classification thresholds, regardless of class imbalance.`

**Review by jeffcarp — COMMENTED** [review]
Thank you so much for the contribution @nikolasavic3! Left some nits, otherwise LGTM

File: src/metrax/metrics.py (hunk 3 above)
Comment (nikolasavic3) [reply src/metrax/metrics.py]: Thank you for reviewing!

**Comment by jeffcarp** [comment]
Thanks again for the contribution!
